tab-oneipl/minutes/2021_12_16_Minutes.rst

```diff
@@ -0,0 +1,147 @@
+* Image data abstraction in oneIPL
+* Memory model in oneIPL
+* Closing words and next plans on oneIPL TAB
+
+
+oneIPL specification walk-through:
+
+* The latest version of oneIPL Spec is published on `oneIPL Spec
+  page`_.
+
+* oneAPI Image Processing Library (oneIPL) consists of several domains,
+  which includes Basic functionality, Color conversion operations, Filtering,
+  Geometry related functionality and potentially can be extended to 3D
+  operations. Current version of specification covers the most important parts
+  of domains and might be extended in future versions.
+
+* The oneIPL programming language is SYCL 2020 based on C++17
+  oneIPL primitives include class data abstractions and functional API
+
+* ipl::image class specifies image data, layout and supported data types,
+  which are defined at compile-time. Each algorithm in the specification
+  contains the layout and data types support matrix. This matrix describes
+  generic layouts as channel count – rows (1,3,4 channels), and generic
+  data types.
+
+* Generic formats are usually supported for multiple devices. Some data
+  types are device specific.
+
+* oneIPL APIs follows SYCL xPU ideology. Each API is able to execute on the
+  range of the devices, if device is supporting required features. oneIPL
+  algorithm shall not substitute unsupported type by the different supported
+  type if it impacts the result. If the type is not supported on the device,
+  the error is fired. SYCL has such checks in kernels as runtime exception.
+  oneIPL can perform such check before, since queue provides information on
+  device features, and spec specifies error conditions.
+
+* Execution mode is asynchronous. Algorithms submitted to sycl::queue and
+  returns the control flow. Execution is scheduled by runtime taking into
+  account the dependencies vector. For arguments having type ipl::image
+  dependencies are handled automatically (example slide 12).
+
+* Example with splitting pipeline. Two functions taking single input and
+  multiple outputs can be executed in parallel, since calls are asynchronous.
+
+* More complex pipeline with watermarking (slide 14) – user-provided kernels
+  like calculating overlay and blending. Some stages can produce output
+  required to host, so shared memory allocator shall be specified in this case.
+  If the image is pure device all intermediate output shall have device
+  allocator.
+
+* oneIPL Image Abstraction. Image can be potentially implemented over host,
+  shared, device and special texture memory. From implementation perspective
+  first three can be done via USM (SYCL2020), texture can be implemented
+  via sycl::image.
+
+* Memory model. oneapi::ipl::image class is basic data abstraction for image
+  data. oneIPL provides single abstraction over different memory types.
+  oneIPL supports different types of memory – device, host, shared and special
+  GPU texture (image) memory.
+
+* See more details on `oneIPL Architecture page`_
```

Comment:
Extra symbol at the end of line

Reply:
Dou you mean "page`_" ?
This is not extra symbol, this is format of links in rst syntax

Reply:
ok, understand